great_expectations/compatibility/pydantic.py

```diff
@@ -0,0 +1,56 @@
+import pydantic
+
+from great_expectations.compatibility.not_imported import (
+    is_version_greater_or_equal,
+)
+
+if is_version_greater_or_equal(version=pydantic.VERSION, compare_version="2.0.0"):
+    # TODO: don't use star imports
+    from pydantic.v1 import *  # noqa: F403
+    from pydantic.v1 import (
+        AnyUrl,
+        UrlError,
+        error_wrappers,
+        errors,
+        fields,
+        generics,
+        json,
+        networks,
+        schema,
+        typing,
+    )
```

Comment:
How does it work if I'm creating a V2 Base Model and I want to use one of these classes? I guess we can cross that bridge when we get there.

Reply:
That won't be possible yet, without some conditional imports.

We would need some kind of `pydantic_v2` compatibility module that works like our other optional dependencies (`sqlalchemy`).
Unless we just decide to drop pydantic v1.